src/transport.js

```diff
@@ -53,11 +54,18 @@ module.exports = function (swarm) {
       // specific for the transport we are using
       const proxyConn = new Connection()
 
-      next(multiaddrs.shift())
+      each(multiaddrs, next)
 
-      // TODO improve in the future to make all the dials in paralell
+      var success = false
       function next (multiaddr) {
+        if (success) {
+          return;
+        }
         const conn = t.dial(multiaddr, () => {
+          if (success) {
+            return;
```

Comment:
Thanks! As it turns out, my each solution is failing in a few cases already (when one address fails immeadiately instead of slowly timing out)